lib/spack/spack/modules/tcl.py

```diff
@@ -20,32 +20,34 @@
 
 
 #: TCL specific part of the configuration
-def configuration():
-    return spack.config.get('modules:tcl', {})
+def configuration(module_set_name):
+    config_path = 'modules:%s:tcl' % module_set_name
+    return spack.config.get(config_path, {})
 
 
 #: Caches the configuration {spec_hash: configuration}
 configuration_registry = {}  # type: Dict[str, Any]
 
 
-def make_configuration(spec):
+def make_configuration(spec, module_set_name):
     """Returns the tcl configuration for spec"""
-    key = spec.dag_hash()
+    key = (spec.dag_hash(), module_set_name)
     try:
         return configuration_registry[key]
     except KeyError:
-        return configuration_registry.setdefault(key, TclConfiguration(spec))
+        return configuration_registry.setdefault(
+            key, TclConfiguration(spec, module_set_name))
 
 
-def make_layout(spec):
+def make_layout(spec, module_set_name):
     """Returns the layout information for spec """
-    conf = make_configuration(spec)
+    conf = make_configuration(spec, module_set_name)
     return TclFileLayout(conf)
 
 
-def make_context(spec):
+def make_context(spec, module_set_name):
```

Comment:
I feel like `make_context`, `make_layout`, and `make_configuration` could be consolidated.  Can the context and the layout not go in the configuration?

Reply:
In terms of logical possibility, yes. Code-structure-wise, the context is an object that owns the layout and the configuration. I think reducing the complexity of this code is a separate PR if we want to take that on.